crates/storage/engines/api.rs

```diff
@@ -43,46 +43,52 @@ pub trait StoreEngine: Debug + Send {
     /// Add block body
     fn add_block_body(
         &mut self,
-        block_number: BlockNumber,
+        block_hash: BlockHash,
         block_body: BlockBody,
     ) -> Result<(), StoreError>;
 
-    /// Obtain block body
+    /// Obtain canonical block body
     fn get_block_body(&self, block_number: BlockNumber) -> Result<Option<BlockBody>, StoreError>;
 
-    /// Add block body
+    /// Obtain any block body using the hash
+    fn get_block_body_by_hash(
```

Comment:
I suspect that callers of this function want the whole block? Should we `have a get_block_by_hash`?

Reply:
Didn't find these cases, but I'll add this in a separate PR as it seems like a low hanging fruit!

Reply:
Actually found this in `transaction.rs`, added issue #422. Adding it right now.

Reply:
Added this in this PR.